JASP-Desktop/widgets/boundqmlcomponentslist.cpp

```diff
@@ -78,20 +100,68 @@ void BoundQMLComponentsList::bindTo(Option *option)
 Option* BoundQMLComponentsList::createOption()
 {
 	Options* templote = new Options();
-	templote->add(_optionKeyName, new OptionVariable());
+
+	if (_termsModel->areTermsInteractions())
+		templote->add(_optionKeyName, new OptionTerm());
+	else
+		templote->add(_optionKeyName, new OptionVariable());
 	addRowComponentsDefaultOptions(templote);
 
 	OptionsTable* result = new OptionsTable(templote);
 	std::vector<Options*> allOptions;
 
-	Terms initTerms = _termsModel->getSourceTerms();
-	for (const Term& term : initTerms)
+	if (hasSource())
+	{
+		Terms initTerms = _termsModel->getSourceTerms();
+		for (const Term& term : initTerms)
+		{
+			Options* row = dynamic_cast<Options*>(templote->clone());
+			if (_termsModel->areTermsInteractions())
+			{
+				OptionTerm* optionVar = new OptionTerm();
+				optionVar->setValue(term.scomponents());
+				row->add(_optionKeyName, optionVar);
+			}
+			else
+			{
+				OptionVariables* optionVar = new OptionVariable();
+				optionVar->setValue(term.asString());
+				row->add(_optionKeyName, optionVar);
+			}
+			allOptions.push_back(row);
+		}
+	}
+	else
 	{
-		Options* row = dynamic_cast<Options*>(templote->clone());
-		OptionVariable* optionVar = new OptionVariable();
-		optionVar->setValue(term.asString());
-		row->add(_optionKeyName, optionVar);
-		allOptions.push_back(row);
+		QVariant defaultValuesVar = getItemProperty("defaultValues");
+		QList<QVariant> defaultValues = defaultValuesVar.toList();
+
+		for (const QVariant& defaultValue : defaultValues)
+		{
+			Options* row = dynamic_cast<Options*>(templote->clone());
+			OptionVariables* optionVar = new OptionVariable();
+			optionVar->setValue(_incrementCounter().toStdString());
+			row->add(_optionKeyName, optionVar);
+
+			QVariantMap defaultValueMap = defaultValue.toMap();
+			QMapIterator<QString, QVariant> it(defaultValueMap);
+			while (it.hasNext())
+			{
+				it.next();
+				QString name = it.key();
+				QVariant value = it.value();
+
+				Option* option = row->get(name.toStdString());
+				if (option)
```

Comment:
`if(option) option->set(Json::Value(value.toJsonValue().toString().toStdString());
else Log::log() << "Default value with name " << name << " is unknown in the component" << std::endl;`